File: 0350/main.py
from typing import Dict, List


# [4,9,5]
# [9,4,9,8,4]

class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        if len(nums1) > len(nums2):
            return self.intersect(nums2, nums1)
        
        hashmap: Dict[int,int] = {}
        value: int = None
        count: int = None

        for value in nums1:
            if (count := hashmap.get(value)) != None:
                hashmap[value] = count + 1
            else:
                hashmap[value] = 1

        res: List[int] = []
        for value in nums2:
            if (count := hashmap.get(value)) != None and count > 0:
                res.append(value)
                hashmap[value] = count - 1       
        return res

File: 0350/main.swift
class Solution {
    func intersect(_ nums1: [Int], _ nums2: [Int]) -> [Int] {
        if nums1.count > nums2.count{
            return intersect(nums2, nums1)
        }

        var hashmap = [Int: Int]()
        for value in nums1{
            if let count = hashmap[value], count > 0{
                hashmap[value] = count + 1
            } else {
                hashmap[value] = 1
            }
        }

        var res = [Int]()
        for value  in nums2{
            if let count = hashmap[value], count > 0{
                res.append(value)
                hashmap[value] = count - 1
            }
        }
        return res
    }
}